src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,26 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+	
+	public String value;
+    protected boolean isPrivate;
+    
+    public Contact(String value, boolean isPrivate) {
+    	this.value = value;
```

Comment:
You may also extract another method from the child classes. One that takes in the regex and then perform the match on the value.